### PR TITLE
Update TiKV store status information

### PR DESCRIPTION
**Pull request by en-jin19**
### What is changed, added or deleted? (Required)
Added the relationship between a TiKV store in "faq/deploy-and-maintain-faq.md", "pd-control.md", and "tidb-scheduling.md".

### Which TiDB version(s) do your changes apply to? (Required)

Although most of Chinese PRs are only applied to master, newly updated PR [#8723](https://github.com/pingcap/docs-cn/pull/8723) updated the contents from v4.0 to master, for consistency, this PR will be also applied to the corresponding versions.

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
1. For the "faq/deploy-and-maintain-faq.md", it is translated from https://github.com/pingcap/docs-cn/pull/7338.
2. For the "pd-control.md", it is translated from https://github.com/pingcap/docs-cn/pull/7177.
3. For the "tidb-scheduling.md", it is translated from https://github.com/pingcap/docs-cn/pull/7338 and https://github.com/pingcap/docs-cn/pull/8723.
4. For the newly added image, it is translated from https://github.com/pingcap/docs-cn/pull/7177.
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by en-jin19** [comment]
@nolouch @TomShawn could you please which versions does this content affect?

**Review by nolouch — APPROVED** [review]

File: pd-control.md [hunk 1]
```diff
@@ -841,6 +841,7 @@ Usage:
 > **Note:**
 >
 > When you use the `store limit` command, the original `region-add` and `region-remove` are deprecated. Use `add-peer` and `remove-peer` instead.
+> You can use PD Control to check the status information (Up, Disconnect, Offline, Down, or Tombstone) of a TiKV store. To see the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).
```
Comment (TomShawn) [inline pd-control.md:844]:
```suggestion
> - When you use the `store limit` command, the original `region-add` and `region-remove` are deprecated. Use `add-peer` and `remove-peer` instead.
> - You can use PD Control to check the status information (Up, Disconnect, Offline, Down, or Tombstone) of a TiKV store. To see the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).
```

**Comment by lilin90** [comment]
> @nolouch @TomShawn could you please which versions does this content affect?

@en-jin19 Has this been confirmed? If not, please further confirm the updates and try to get this PR merged.

**Comment by lilin90** [comment]
@en-jin19 And please fix the checkbox syntax in the PR description box.
image

File: faq/deploy-and-maintain-faq.md [hunk 2]
```diff
@@ -266,6 +266,10 @@ PD can tolerate any synchronization error, but a larger error value means a larg
 
 The client connection can only access the cluster through TiDB. TiDB connects PD and TiKV. PD and TiKV are transparent to the client. When TiDB connects to any PD, the PD tells TiDB who is the current leader. If this PD is not the leader, TiDB reconnects to the leader PD.
 
+#### What is the relationship between each status (Up, Disconnect, Offline, Down, Tombstone) of a TiKV store?
+
+You can use PD Control to check the status information of a TiKV store. To see the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).
```
Comment (en-jin19) [inline faq/deploy-and-maintain-faq.md:271]:
```suggestion
You can use PD Control to check the status information of a TiKV store. For the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).
```

File: tidb-scheduling.md [hunk 3]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
     * Whether the store is overloaded
     * Labels (See [Perception of Topology](/schedule-replicas-by-topology-labels.md))
 
+You can use PD control to check the status information of a TiKV store, which is divided into Up, Disconnect, Offline, Down, and Tombstone. The relationship between each status is as follows:
+
++ **Up**: The current TiKV store is providing service.
++ **Disconnect**: When the heartbeat information of PD and the TiKV store is lost for more than 20 seconds, the store status changes to Disconnect. When the lost time exceeds the time specified by `max-store-down-time`, the store status changes to Down.
++ **Down**: When the time that the TiKV store lost connection with the cluster has exceeded the time specified by `max-store-down-time` (30 minutes by default), the store changes to this status. This status indicates that the corresponding store status changes to Down and starts replenishing peers of each Region on the surviving store.
```
Comment (en-jin19) [inline tidb-scheduling.md:84]:
```suggestion
+ **Down**: When the time that the TiKV store lost connection with the cluster is longer than the time specified by `max-store-down-time` (30 minutes by default), the store changes to "Down". In this status, the store starts replenishing peers of each Region on the surviving store.
```

File: tidb-scheduling.md [hunk 4]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
...
++ **Down**: When the time that the TiKV store lost connection with the cluster is longer than the time specified by `max-store-down-time` (30 minutes by default), the store changes to "Down". In this status, the store starts replenishing peers of each Region on the surviving store.
++ **Offline**: If you manually take a TiKV store offline through PD Control, the store status changes to "Offline". This is only an intermediate status that the store is taking offline. The store in this status performs the operations of leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows that both operations of leader transfer and Region balance have been completed, the store status changes to "Tombstone" from "Offline". In the "Offline" status, you should **not** disable the store service and the physical server where the store is located.
++ **Tombstone**: This status indicates that the TiKV store is completely offline. You can use `remove-tombstone` interface to safely clean up TiKV in this status.
```
Comment (shichun-0415) [inline tidb-scheduling.md:86]:
```suggestion
+ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
+ **Down**: Heartbeat messages between the PD and the TiKV store are lost for a time longer than `max-store-down-time` (30 minutes by default). In this status, the TiKV store starts replenishing replicas of each Region on the surviving store.
+ **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.
+ **Tombstone**: The TiKV store is completely offline. You can use the `remove-tombstone` interface to safely clean up TiKVs in this status.
```

File: faq/deploy-and-maintain-faq.md [hunk 5]
```diff
@@ -266,6 +266,10 @@ PD can tolerate any synchronization error, but a larger error value means a larg
 
 The client connection can only access the cluster through TiDB. TiDB connects PD and TiKV. PD and TiKV are transparent to the client. When TiDB connects to any PD, the PD tells TiDB who is the current leader. If this PD is not the leader, TiDB reconnects to the leader PD.
 
+#### What is the relationship between each status (Up, Disconnect, Offline, Down, Tombstone) of a TiKV store?
+
+You can use PD Control to check the status information of a TiKV store. For the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).
```
[the remark this answers is not in the document]
Comment (en-jin19) [reply faq/deploy-and-maintain-faq.md:271]: 这个图表示的各状态之间的关系其实不是特别清晰的，适合用作为辅助材料。

另外，为了直接回答标题问题，感觉可以把两句的顺序调换过来。毕竟如果解释了关系，可能也有人会好奇那要看这些关系需要怎么操作。

Comment (en-jin19) [inline faq/deploy-and-maintain-faq.md:271]:
```suggestion
For the relationship between each status, refer to [Relationship between each status of a TiKV store](/tidb-scheduling.md#information-collection).

You can use PD Control to check the status information of a TiKV store. 
```

File: pd-control.md [hunk 6]
```diff
@@ -840,7 +840,8 @@ Usage:
 
 > **Note:**
 >
-> When you use the `store limit` command, the original `region-add` and `region-remove` are deprecated. Use `add-peer` and `remove-peer` instead.
+> - When you use the `store limit` command, the original `region-add` and `region-remove` are deprecated. Use `add-peer` and `remove-peer` instead.
```
[the remark this answers is not in the document]
Comment (en-jin19) [reply pd-control.md:843]: 这部分在本 PR 中没有修改，不过对于这部分，我的理解是指当使用 `store limit` command 时参数才会被废弃，所以保留原句，表示时间条件如何？

File: tidb-scheduling.md [hunk 7]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
     * Whether the store is overloaded
     * Labels (See [Perception of Topology](/schedule-replicas-by-topology-labels.md))
 
+You can use PD control to check the status information of a TiKV store, which is divided into Up, Disconnect, Offline, Down, and Tombstone. The relationship between each status is as follows:
```
[the remark this answers is not in the document]
Comment (en-jin19) [reply tidb-scheduling.md:80]: > 下面更多是在分别解释每个状态的含义，并不是明确的 the relationship between each status。

好哒，下面除了 UP 之外每个状态同时包含了两个内容：对各状态的描述，在哪些情况下能从一个状态切换至另一个状态。另外其他文章中也把 the relationship between each status 链到这里了，所以我修改了在上面描述，同时反应这两句，PTAL， 感谢！

[the remark this answers is not in the document]
Comment (shichun-0415) [reply tidb-scheduling.md:80]: > The following is the description of each status and the relationship between each of them

OK, 没问题，最后那句可以调整下表达：The following is a description of all statuses and their relationship.

File: pd-control.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (shichun-0415) [reply pd-control.md:843]: @nolouch PTAL. Thanks.

File: tidb-scheduling.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (en-jin19) [reply tidb-scheduling.md:80]: > > The following is the description of each status and the relationship between each of them
> 
> OK, 没问题，最后那句可以调整下表达：The following is a description of all statuses and their relationship.

谢谢，已修改！

File: pd-control.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JmPotato) [reply pd-control.md:843]: `region-add` 和 `region-remove` 这两个概念确实只存在与跟 store-limit 相关的使用上，但我觉得还是 @shichun-0415 修改的这个说法更严谨吧，毕竟从各种意义上来说原有的这两个概念都不存在了并且被换成了 `add-peer` 和 `remove-peer`，不加条件语句更好。

File: tidb-scheduling.md [hunk 8]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
...
 
+You can use PD control to check the status of a TiKV store, which can be Up, Disconnect, Offline, Down, or Tombstone. The following is a description of all statuses and their relationship.
+
++ **Up**: The current TiKV store is in service.
++ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
++ **Down**: Heartbeat messages between the PD and the TiKV store are lost for a time longer than `max-store-down-time` (30 minutes by default). In this status, the TiKV store starts replenishing replicas of each Region on the surviving store.
++ **Offline**: A TiKV store is manually taken offline through PD Control. In this status, the TiKV store is intermediately offline and performs leader transfer and Region balance. When the `leader_count/region_count` (obtained through PD Control) shows completion of leader transfer and Region balance, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service and the physical server where the store is located.
```
Comment (JmPotato) [inline tidb-scheduling.md:85]: `leader transfer and Region balance` 的说法不准确，感觉中文原文也得改一下，我一会在中文文档那边提个 PR 吧。事实上通过非部署工具触发下线的时候，既不会发生 leader transfer 也不会发生 region balance，准确的说法是“会开始将下线节点上面的 Region 搬离下线 TiKV Store”，而且是有条件限制的。Region balance 在 TiKV 集群里有其他的概念，容易产生歧义。

[the remark this answers is not in the document]
Comment (en-jin19) [reply tidb-scheduling.md:85]: @JmPotato @shichun-0415 this part is revised, PTAL, thank you!

[the remark this answers is not in the document]
Comment (JmPotato) [reply tidb-scheduling.md:85]: It should be `or` here.

> do not disable the store service **_or_** the physical server where the store is located.

The rest LGTM.

[the remark this answers is not in the document]
Comment (shichun-0415) [reply tidb-scheduling.md:85]: Suggestion: The store in this status moves all its Regions to other "Up" stores that meet the relocation conditions.
> The store in this status moves all Regions from this store to other stores that meet the relocation conditions and are in the "Up" status at the same time.

Suggestion: During the process that the store goes offline, if the cluster does not have target stores to relocate the Regions (for example, inadequate stores to hold replicas in the cluster), the store is always in the "Offline" status.
> During the process that the store goes offline, if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.

File: tidb-scheduling.md [hunk 9]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
...
++ **Down**: Heartbeat messages between the PD and the TiKV store are lost for a time longer than `max-store-down-time` (30 minutes by default). In this status, the TiKV store starts replenishing replicas of each Region on the surviving store.
++ **Offline**: A TiKV store is manually taken offline through PD Control. This is only an intermediate status for the store to go offline. The store in this status moves all its Regions to other "Up" stores that meet the relocation conditions. When `leader_count` and `region_count` (obtained through PD Control) both show `0`, the store status changes to "Tombstone" from "Offline". In the "Offline" status, **do not** disable the store service or the physical server where the store is located. During the process that the store goes offline, if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.
```
Comment (en-jin19) [inline tidb-scheduling.md:85]: @JmPotato @shichun-0415 all comments are addressed, PTAL, thank you!

Comment (shichun-0415) [reply tidb-scheduling.md:85]: if the cluster does not have other target stores that meet the relocation conditions (for example, there are not enough stores that can meet the requirements for the number of replicas in the cluster anymore), the store always is in the "Offline" status.---**Seems this sentence is still the original one.**

**Review by JmPotato — APPROVED** [review]
LGTM!

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: tidb-scheduling.md [hunk 10]
```diff
@@ -77,6 +77,16 @@ Scheduling is based on information collection. In short, the PD scheduling compo
     * Whether the store is overloaded
     * Labels (See [Perception of Topology](/schedule-replicas-by-topology-labels.md))
 
+You can use PD control to check the status of a TiKV store, which can be Up, Disconnect, Offline, Down, or Tombstone. The following is a description of all statuses and their relationship.
+
++ **Up**: The current TiKV store is in service.
++ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
```
Comment (en-jin19) [inline tidb-scheduling.md:83]:
```suggestion
+ **Up**: The TiKV store is in service.
+ **Disconnect**: Heartbeat messages between the PD and the TiKV store are lost for more than 20 seconds. If the lost period exceeds the time specified by `max-store-down-time`, the status changes to "Down".
```

**Pull request merged**